src/PKSim.Presentation/Presenters/ExpressionProfiles/ExpressionProfileMoleculesPresenter.cs

```diff
@@ -57,30 +63,49 @@ public void Edit(ExpressionProfile expressionProfile)
          activateMoleculeExpressionPresenter();
       }
 
+      public void Save()
+      {
+         _expressionProfile.Category = _expressionProfileDTO.Category;
+         _expressionProfileUpdater.UpdateMoleculeName(_expressionProfile, _expressionProfileDTO.MoleculeName);
+      }
+
       public void SpeciesChanged()
       {
          _expressionProfileFactory.UpdateSpecies(_expressionProfile, _expressionProfileDTO.Species);
          refreshExpression();
       }
 
-      public void CategoryChanged()
+      public bool IsEditMode
       {
-         _expressionProfile.Category = _expressionProfileDTO.Category;
-      }
+         get => _isEditMode;
+         set
+         {
+            _isEditMode = value;
+            //Edit vs Create. We can only change naming settings when CREATING an Expression Profile
+            if(_isEditMode)
+               _view.DisableSettings();
 
-      public void MoleculeNameChanged()
-      {
-         _expressionProfileUpdater.UpdateMoleculeName(_expressionProfile, _expressionProfileDTO.MoleculeName);
+         }
       }
 
-      public void DisableSettings() => _view.DisableSettings();
-
+      
       public void LoadExpressionFromDatabaseQuery()
       {
-         if (!_editMoleculeTask.CanQueryProteinExpressionsFor(_expressionProfile.Individual))
+         if (!_expressionProfileProteinDatabaseTask.CanQueryProteinExpressionsFor(_expressionProfile))
             throw new OSPSuiteException(PKSimConstants.Error.NoProteinExpressionDatabaseAssociatedTo(_expressionProfile.Species.Name));
 
-         AddCommand(_editMoleculeTask.EditMolecule(_expressionProfile.Molecule, _expressionProfile.Individual, _expressionProfileDTO.MoleculeName));
+         var queryResults = _expressionProfileProteinDatabaseTask.QueryDatabase(_expressionProfile, _expressionProfileDTO.MoleculeName);
+         if (queryResults == null)
+            return;
+
+         if (!IsEditMode)
+         {
+            _expressionProfileDTO.MoleculeName = queryResults.ProteinName;
+            _moleculeParameterTask.SetDefaultFor(_expressionProfile, _expressionProfileDTO.MoleculeName);
```

Comment:
This is a bit of a breaking change compared to before. Entering the name CYP3A4 without using the DB will not set the default value...I am not sure how to do this

Reply:
You mean that if you e.g. set the name = CYP3A4 **manually**, then ReferenceConcentration, HalfTimes etc. of CYP3A4 will not be used?

Reply:
Yes this is the problem. Before the input was a modal so we knew what to set